usr/src/uts/common/fs/zfs/dsl_pool.c

```diff
@@ -129,6 +130,17 @@ uint64_t zfs_delay_scale = 1000 * 1000 * 1000 / 2000;
 hrtime_t zfs_throttle_delay = MSEC2NSEC(10);
 hrtime_t zfs_throttle_resolution = MSEC2NSEC(10);
 
+
+/*
+ * Tunable to control number of threads servicing the vn rele taskq
+ */
+int zfs_vn_rele_threads = 256;
```

Comment:
that's what I thought too, and with TQ_SLEEP that seems to hold true.
However I've switched to TQ_NOSLEEP and now I'm seeing only get 2-3 threads created even when deleting 6 files!?